tfx_bsl/beam/run_inference.py

```diff
@@ -96,18 +98,126 @@ class OperationType(object):
 
 
 @beam.ptransform_fn
-@beam.typehints.with_input_types(Union[tf.train.Example,
-                                       tf.train.SequenceExample])
+@beam.typehints.with_input_types(tf.train.Example)
 @beam.typehints.with_output_types(prediction_log_pb2.PredictionLog)
-def RunInferenceImpl(  # pylint: disable=invalid-name
+def RunInferenceOnExamples(  # pylint: disable=invalid-name
     examples: beam.pvalue.PCollection,
-    inference_spec_type: model_spec_pb2.InferenceSpecType
+    inference_spec_type: model_spec_pb2.InferenceSpecType,
+    schema: Optional[schema_pb2.Schema] = None
 ) -> beam.pvalue.PCollection:
-  """Implementation of RunInference API.
+  """Run inference with a model.
+
+   There are two types of inference you can perform using this PTransform:
+   1. In-process inference from a SavedModel instance. Used when
+     `saved_model_spec` field is set in `inference_spec_type`.
+   2. Remote inference by using a service endpoint. Used when
+     `ai_platform_prediction_model_spec` field is set in
+     `inference_spec_type`.
+
+   TODO(b/131873699): Add support for the following features:
+   1. Bytes as Input.
+   2. PTable Input.
+   3. Models as SideInput.
 
   Args:
     examples: A PCollection containing examples.
     inference_spec_type: Model inference endpoint.
+    Schema [optional]: required for models that requires
+      multi-tensor inputs.
+
+  Returns:
+    A PCollection containing prediction logs.
+  """
+
+  data_type = DataType.EXAMPLE
+  converter = tf_example_record.TFExampleBeamRecord(
+    physical_format="inmem",
+    telemetry_descriptors=[],
```

Comment:
physical_format could be "on_wire"
telemetry_descriptors are [_METRICS_DESCRIPTOR_INFERENCE, operation_type, proximity_descriptor] like here: https://github.com/tensorflow/tfx-bsl/blob/master/tfx_bsl/beam/run_inference.py#L228